### PR TITLE
Convenience method for tonight

### PR DESCRIPTION
**Pull request by wtgee**
This is a simple convenience method to return the time values for `tonight`. If already night it returns `Time.now()` as the first item.

I wasn't sure what the best way to write a test for this would be but if interested in this addition I'd be happy to do something. Seems like a base convenience method a lot of people would need.

### CODE REVIEW

**Comment by wtgee** [comment]
Seems to be failing on our good friend IERS_A...

File: astroplan/observer.py [hunk 1]
```diff
@@ -1663,3 +1663,35 @@ def target_hour_angle(self, time, target):
             hour_angle = Longitude(self.local_sidereal_time(time) - coord.ra)
 
         return hour_angle
+
+    @u.quantity_input(horizon=u.deg)
+    def tonight(self, horizon=0 * u.degree):
+        """
+        Return a time range corresponding to the nearest night
+
+        This will return a range of `Time` values from the `astronomical_twilight`
```
Comment (bsipocz) [inline astroplan/observer.py]: You need to add the full namespace for the functions/classes to make sphinx recognize it. e.g. `~astropy.time.Time`. The ``~`` in the front means that the path will be hidden in the rendered documentation. 
(Need to do it for ``astronomical_twilight`` and ``astronomical_sunrise``, too)

**Comment by bsipocz** [comment]
Could you please rebase it, so only the relevant commits remain (no upsteam merges, etc.).

**Comment by wtgee** [comment]
I think I made that worse...too many awake hours, sorry.

**Comment by bsipocz** [comment]
Tiny bit, but at least the tests pass ;) I guess part of the mess is that you work in your master branch and not a separate feature one. So when you update the master it's not possible to make a clean fast forward pull.

You can say that ``git rebase -i upstream/master``. It comes up with an editor.
And delete everything but the last three commits  (96bfb0d, 27bab5f, f1bd173) and and save it. It should go without conflicts.
Then you need to force push.

**Comment by bmorris3** [comment]
This is great @wtgee, I've implemented this probably ten times in my code, I'd be happy to see it built into `Observer`.

First, you should follow @bsipocz's rebase advise so your early commits that are superseded by other merged PRs go away.

I think it would be useful for some users and for testing if the first keyword argument was `time=None` – then you set `current_time = Time.now() if time is None else time`. That way we can pass test times to the `time` kwarg.

We do need some tests to prove it's working. You should make a new test function in `astroplan/tests/test_observer.py` called `test_tonight`. Then you should which checks that the input and output times are as you would expect them for (1) an input time in the middle of a night and (2) an input time during the day. 

You can check for (approximate) equivalence of the times you'd expect using [this format used in other time tests](https://github.com/astropy/astroplan/blob/master/astroplan/tests/test_observer.py#L299-L334). You don't have to use PyEphem to generate the sunrise/sunset times like we did in that example.

**Comment by wtgee** [comment]
Thanks @bmorris3, I'll get some tests written up and add in the `time` argument. The code also needs a `which` to get the next sunrise/set. I'll fix the rebase and write up some tests a little bit later today.

**Comment by wtgee** [comment]
Moved to #204 as a non-cluttered branch.

**Pull request closed without merging**